### PR TITLE
DOC: release-versioned links to GH notebooks

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request adds support in our sphinx build to point to release-versioned GH notebooks, and links to the example notebooks for each of the respective viztools in a new quickstart section (along with a link to download the sample data and a snippet of how to launch in the standalone voila app).

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: docs/mosviz/index.rst [hunk 1]
```diff
@@ -12,6 +12,17 @@ Mosviz is a quick-look analysis and visualization tool for multi-object spectros
 
 The NIRSpec micro-shutter array (MSA) will produce ~100 spectra per pointing. Many users will perform surveys with the MSA that will result in data sets containing many spectra. This tool allows users to inspect the locations of astronomical sources within shutters, the location of background apertures in the observed field, the quality of the 2D spectra, and the quality of the 1D extracted spectra. It also often involves simple interactive measurements of quantities such as wavelengths, velocities, line fluxes, widths.
 
+Quickstart
+----------
+
+To load a sample `NIRISS MOS Data Set <https://stsci.box.com/shared/static/9lkf5zha6zkf8ujnairy6krobbh038wt.zip>`_ into ``Mosviz`` in the standalone app, unzip the downloaded zip file and run::
+
+    jdaviz mosviz /path/to/NIRISS_for_parser_p0171
```
[the remark this answers is not in the document]
Comment (duytnguyendtn) [reply docs/mosviz/index.rst:20]: Summarizing our conversation we had off Github: directory loading currently assumes Nirspec, as you can't currently specify an instrument yet (I recall being instructed explicitly to not implement an instrument argument). The above data is for NIRISS, hence why it's probably not working

**Comment by kecnry** [comment]
The idea is that if someone is accessing RTD for a released 2.X version, we don't want to point them to notebooks in the main branch of GitHub that may include features/API not in their installed version (in other words, outdated docs should point to outdated notebooks to avoid confusion).

File: docs/conf.py [hunk 2]
```diff
@@ -149,6 +149,12 @@
 # The short X.Y version.
 version = '.'.join(release.split('.')[:2])
 
+extensions += ['sphinx.ext.extlinks']
+gh_tag = f'v{release}' if '.dev' not in release else 'main'
+extlinks = {'gh-tree': (f'https://github.com/spacetelescope/jdaviz/tree/{gh_tag}/%s', '%s'),
+            'gh-notebook': (f'https://github.com/spacetelescope/jdaviz/blob/{gh_tag}/notebooks/%s.ipynb',
+                            '%s notebook')}
```
[the remark this answers is not in the document]
Comment (pllim) [reply docs/conf.py:156]: Theoretically, you could setup your own RTD for your Jdaviz fork, and then test out this branch on your fork by applying some fake tags to it.

When the testing is done, you will have to clean up your fork but disabling the fork RTD and deleting the fake tags.

Or we can YOLO and see how it working in the next release.

We don't have to use `stable` at all if we are using release tags. `latest` will always render from `main`.

[the remark this answers is not in the document]
Comment (kecnry) [reply docs/conf.py:156]: Sounds messy... I'm all for YOLOing then! I did manually override the `release` variable immediately before these lines and the link did render as expected and point to the correct version on GitHub, so I think it should work fine (famous last words).

File: docs/cubeviz/index.rst [hunk 3]
```diff
@@ -51,6 +51,16 @@ Future functionality will include the ability to:
 
 * bin data into constant signal-to-noise regions
 
+Quickstart
```
[the remark this answers is not in the document]
Comment (kecnry) [reply docs/cubeviz/index.rst:54]: `:gh-notebook:CubevizExample` uses the new configuration in `conf.py` to point to the correct URL on GitHub for `CubevizExample.ipynb`.  The goal here is to include links to our example notebooks in the docs, and that required support for linking to the correct URL given the release version.

[the remark this answers is not in the document]
Comment (pllim) [reply docs/cubeviz/index.rst:54]: Ah, okay. Thanks for the clarification!

File: docs/conf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pllim) [reply docs/conf.py:156]: p.s. This is real fancy. Nice work!

**Review by rosteen — APPROVED** [review]

**Comment by rosteen** [comment]
My review is a little YOLO-y, but let's get this in.

**Pull request merged**